core/src/main/java/org/apache/carbondata/core/datastore/page/encoding/DefaultEncodingStrategy.java

```diff
@@ -130,7 +146,8 @@ private static DataType fitLongMinMax(long max, long min) {
     }
   }
 
-  private static DataType fitMinMax(DataType dataType, Object max, Object min) {
+  private static DataType fitMinMax(DataType dataType, Object max, Object min,
+      DecimalConverterFactory.DecimalConverterType decimalConverterType) {
```

Comment:
I think it is not good to pass `decimalConverterType` in many functions. It makes code complex. It is better to think of a way to encapsulate it.